operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ConfigMapOperator.java

```diff
@@ -36,6 +40,19 @@ public ConfigMapOperator(Vertx vertx, KubernetesClient client) {
 
     @Override
     protected Future<ReconcileResult<ConfigMap>> internalPatch(String namespace, String name, ConfigMap current, ConfigMap desired) {
+        JsonNode diff = JsonDiff.asJson(patchMapper().valueToTree(current), patchMapper().valueToTree(desired));
+        boolean metricsChanged = false;
+        for (JsonNode d : diff) {
+            if (d.get("path").asText().equals("/data/metrics-config.yml") && d.get("op").asText().equals("replace")) {
```

Comment:
Sorry, I didn't used the JsonDiff before. Does this `d.get("op").asText().equals("replace")` make sure we do the rolling update when metrics were newly added?

Reply:
In that case `d.get("op").asText().equals("update") == true`. I tested this (i hope properly).

Reply:
`ConfigMapOperator` is supposed to be a generic class for all CMs, so you shouldn't go putting something that's specific to CMs used for logging in here. If you need to do that you can subclass it. This is why we have `StatefulSetOperator`, `KafkaSetOperator` and `ZookeeperSetOperator`.